### ISSUE TITLE
[Off-canvas] Foundation 6 is incompatible with Foundation 5 in terms of multiple off-canvas on a single page

### ISSUE THREAD

**Issue by alexeyhimself** [zurb/foundation-sites#7755]
**Steps to reproduce**

Case with 2 different one-side menus, sharing common content:

``` html
<div class="off-canvas-wrapper">
    <div class="off-canvas-wrapper-inner" data-off-canvas-wrapper>
      <div class="off-canvas position-left" id="offCanvasLeft1" data-off-canvas>Left1</div>
      <div class="off-canvas position-left" id="offCanvasLeft2" data-off-canvas>Left2</div>
      <div class="off-canvas-content" data-off-canvas-content>Common content</div>
    </div>
  </div>
<button type="button" class="button" data-toggle="offCanvasLeft1">Open Left1</button>
<button type="button" class="button" data-toggle="offCanvasLeft2">Open Left2</button>
```

Case with 2 different one-side menus, sharing different contents:

``` html
<div class="off-canvas-wrapper">
    <div class="off-canvas-wrapper-inner" data-off-canvas-wrapper>
      <div class="off-canvas position-left" id="offCanvasLeft11" data-off-canvas>Left11</div>
      <div class="off-canvas-content" data-off-canvas-content>Content11</div>
    </div>
  </div>
<button type="button" class="button" data-toggle="offCanvasLeft11">Open Left11</button>

<div class="off-canvas-wrapper">
    <div class="off-canvas-wrapper-inner" data-off-canvas-wrapper>
      <div class="off-canvas position-left" id="offCanvasLeft22" data-off-canvas>Left22</div>
      <div class="off-canvas-content" data-off-canvas-content>Content22</div>
    </div>
  </div>
<button type="button" class="button" data-toggle="offCanvasLeft22">Open Left22</button>
```

**Expected behavior**
Should be slided only canvas with id == data-toggle, for example, if clicked button with data-toggle="offCanvasLeft22", then only div with id="offCanvasLeft22" is slided, but not all canvases on the page

**Actual behavior**
All canvases slide at the same time, no matter if data-toggle == id matched or not.

**Foundation 5**
In Foundation 5 was <aside> element that allowed clear mapping and such implementations. And even nested implementations.

Now I can't move from Foundation 5 to Foundation 6.
Please, consider this issue, if it fits your vision.
Thank you!

P.s.: If you're curious about implementation it could be used, then take a look at a gif: https://s3-eu-west-1.amazonaws.com/sqatmt/img/kick-ass-testcases-editor.gif - here used and demonstrated 2 off-canvas menues hidden by 2 blue vertical splitters. Implemented with Foundation 5.

**Comment by alexeyhimself** [comment]
For any case, if you are not going to fix this, I post rough, but working workarounds for both cases:

Workaround for a case with 2 different one-side menus, sharing different contents:
```html
<div class="off-canvas-wrapper">
    <div id="offCanvasLeft11" class="off-canvas-wrapper-inner" data-off-canvas-wrapper>
      <div class="off-canvas position-left" data-off-canvas>Left11</div>
      <div class="off-canvas-content" data-off-canvas-content>Content11</div>
    </div>
  </div>
<button type="button" class="button" onclick="$('#offCanvasLeft11').toggleClass('is-off-canvas-open is-open-left');">Open Left11</button>

<div class="off-canvas-wrapper">
    <div id="offCanvasLeft22" class="off-canvas-wrapper-inner" data-off-canvas-wrapper>
      <div class="off-canvas position-left" data-off-canvas>Left22</div>
      <div class="off-canvas-content" data-off-canvas-content>Content22</div>
    </div>
  </div>
<button type="button" class="button" onclick="$('#offCanvasLeft22').toggleClass('is-off-canvas-open is-open-left');">Open Left22</button>
```
is to move id from data-off-canvas element to data-off-canvas-wrapper and ask button click to toggle class that responds for displaying menu

Workaround for a case with 2 different one-side menus, sharing common content:
```html
<div class="off-canvas-wrapper">
    <div id="offCanvasLeft" class="off-canvas-wrapper-inner" data-off-canvas-wrapper>
      <div class="off-canvas position-left" id="offCanvasLeft1" data-off-canvas>Left1</div>
      <div class="off-canvas position-left" id="offCanvasLeft2" data-off-canvas>Left2</div>
      <div class="off-canvas-content" data-off-canvas-content>Common content</div>
    </div>
  </div>
<button type="button" class="button" onclick="$('#offCanvasLeft1').show();$('#offCanvasLeft2').hide();$('#offCanvasLeft').toggleClass('is-off-canvas-open is-open-left');">Open Left1</button>
<button type="button" class="button" onclick="$('#offCanvasLeft1').hide();$('#offCanvasLeft2').show();$('#offCanvasLeft').toggleClass('is-off-canvas-open is-open-left');">Open Left2</button>
```
is to assign id on data-off-canvas-wrapper and ask button click to toggle class that responds for displaying menu and hide data-off-canvas

**Comment by rafibomb** [comment]
Thanks for posting your solution! It will definitely be helpful. This change is not on our roadmap for off-canvas but we'll see if it's something we can add without breaking changes or too much code.

**Comment by eysdre** [comment]
good solution!tnx

**Comment by kball** [comment]
Off-canvas is being re-written for 6.3, would be good to see if this is addressed by #9357

**Comment by brettsmason** [comment]
Off canvas has been rewritten for 6.3, and this should work out of the box now - see [this visual test](https://github.com/zurb/foundation-sites/blob/develop/test/visual/offcanvas/absolute.html).
Release candidate is out today, with the full release scheduled for the 15th December.

Closing this, but if you still have issues with 6.3 please re-open.
